app/services/order_purchaser.rb

```diff
@@ -55,15 +55,39 @@ def order_detail_updater
   end
 
   def order_in_the_past!
-    unless order.can_backdate_order_details?
+    unless can_backdate_order_details?
       raise NUCore::PurchaseException.new(I18n.t("controllers.orders.purchase.future_dating_error"))
     end
 
     # update order detail statuses if you've changed it while acting as
     if acting_as? && order_status_id.present?
-      order.backdate_order_details!(user, order_status_id)
-    else
-      order.complete_past_reservations!
+      backdate_order_details!(user, order_status_id)
+    end
+    complete_past_reservations!
+  end
+
+  def can_backdate_order_details?
+    order.ordered_at <= Time.zone.now
+  end
+
+  def backdate_order_details!(update_by, order_status)
+    # can accept either an order status or an id
+    order_status = order_status.is_a?(OrderStatus) ? order_status : OrderStatus.find(order_status)
```

Comment:
Since this is only used in one scenario, can we get rid of this flexibility and just accept either an id or an `OrderStatus` object?